rfcs/0001-internal-error-reporting.md

```diff
@@ -0,0 +1,124 @@
+
+Exception messages are the least-controllable data item here; currently, Seq itself avoids including senstive information in exception messages. By their nature, however, exceptions are unpredictable. Initial implementation of the feature will need to include a review current exception messages, and we will futher need to monitor the content of reported exceptions to limit the amount of user/implementation-specific data that is collected.
+
+### Blacklisted components
+
+Known security-sensitive components, e.g. the authentication providers, will be explicitly excluded from crash reporting (e.g. by pushing `Telemetry = false` on the internal log context and filtering these events from reporting.)
+
+Exceptions generated by plug-in apps will also be excluded, as it's not possible to determine in advance what information might be revealed by a user-created app.
+
+The blacklisted components will be determined by review.
+
+### Internal crash collection/queueing
+
+To avoid destabilizing the Seq server, or consuming excessive memory or bandwidth, the number of queued/reported exceptions needs to be capped. Starting with very small initial limits seems sensible:
+
+* Event size limit - 16 kB
+* Reported exceptions - 100/day, (therefore a maximum 1600 kB per day egress)
+* Queued exceptions internally awaiting transmission - 10
+
+These values can easily be modified in the future, if desired.
+
+### Transport/protocol
+
+The error reports will be sent in JSON format via HTTPS to a remote collection endpoint.
+
+### Data retention
+
+By default, we will retain error reports for 30 days or less. If an error report results in an issue being raised, we will copy non-identifying data to an issue tracker ticket.
+
+### Continuity
+
+The feature must not cause any undesirable performance, data transfer, or reliability side-effects if/when the server-side reporting endpoint is switched off or unavailable.
+
+### API client update
+
+The system settings for crash reporting will need to be exposed by the _Seq.Api_ C# client.
+
+## Drawbacks
+
+Collecting any end-user data introduces privacy and security concerns. The feature design attempts to mitigate these through:
+
+* User opt-in
+* Limited scope of reporting (unhandled errors only)
+* Exclusion of detailed event properties
+* HTTPS transport
+* Blacklisted components
+* (Server-side) short data retention time
+
+## Alternatives
+
+* Reports could be generated and sent from a separate "monitor" process, particularly so that non-.NET exceptions can be collected and reported; the added implementation cost rules this out in the short-term, but it may be useful as a future enhancement
+* Crash telemetry settings could be enabled under _Settings_ > _Diagnostics_; this would be a more logical grouping, but the layout of the _Diagnostics_ page does not incorporate form-style fields
```

Comment:
I'm thinking the _Diagnostics_ page might be a better home for this, since the time users will want to interact with it is probably the same time they're digging through diagnostics.

We could be consistent with the _Apps_ page, which has some settings component at the top, and then a data component at the bottom.

Thoughts?